cstar/roms/component.py

```diff
@@ -471,3 +442,80 @@ def post_run(self) -> None:
                     cwd=out_path,
                     shell=True,
                 )
+
+
+class ROMSDiscretization(Discretization):
+    """
+    An implementation of the Discretization class for ROMS.
+
+
+    Additional attributes:
+    ----------------------
+    n_procs_x,n_procs_y: int
+        The number of processes following the x and y directions, for running in parallel
+
+    Properties:
+    -----------
+    n_procs_tot: int
+        The value of n_procs_x * n_procs_y
+
+    """
+
+    def __init__(
+        self,
+        time_step: int,
+        nx: Optional[int] = None,
+        ny: Optional[int] = None,
+        n_levels: Optional[int] = None,
+        n_procs_x: Optional[int] = None,
+        n_procs_y: Optional[int] = None,
+    ):
+        """
+        Initialize a ROMSDiscretization object from basic discretization parameters
+
+        Parameters:
+        -----------
+        time_step: int
+            The time step with which to run the Component
+        nx,ny,n_levels: int
+            The number of x and y points and vertical levels in the domain associated with this object
+        n_procs_x,n_procs_y: int
+            The number of processes following the x and y directions, for running in parallel
+
+
+        Returns:
+        --------
+        ROMSDiscretization:
+            An initialized ROMSDiscretization object
+
+        """
+
+        super().__init__(time_step, nx, ny, n_levels)
+        self.n_procs_x = n_procs_x
+        self.n_procs_y = n_procs_y
+
+    @property
+    def n_procs_tot(self) -> Optional[int]:
+        """Total number of processors required by this ROMS configuration"""
+        if (self.n_procs_x is None) or (self.n_procs_y is None):
+            return None
+        else:
+            return self.n_procs_x * self.n_procs_y
```

Comment:
How can this be `None`? Surely if you're not running in parallel the default for `n_procs_x`, `n_procs_y`, and `n_procs_tot` would be `1`, not `None`?

Reply:
Good point! Addressed in b69800f70ca3a583ffdde906d663d847df97404b